src/YarpUtilities/src/VectorsCollectionServer.cpp

```diff
@@ -96,14 +96,17 @@ bool VectorsCollectionServer::populateMetadata(const std::string& key,
                                                const std::vector<std::string>& metadata)
 {
     // check if the metadata has been already populated
-    if (m_pimpl->isMetadataFinalized)
-    {
-        log()->error("[VectorsCollectionServer::populateMetadata] The metadata has been already "
-                     "populated.");
-        return false;
-    }
+    // This code was temporarily removed via commenting it out, this will be discussed with the team
+    // later on how to append metadata after it is finalized
+
+    // if (m_pimpl->isMetadataFinalized)
+    // {
+    //     log()->error("[VectorsCollectionServer::populateMetadata] The metadata has been already "
+    //                  "populated.");
+    //     return false;
+    // }
```

Comment:
This needs to be re-added before merging this PR 🤔. Why was it removed in the first place?

Reply:
This would be better to discuss face-to-face. This has to do with the issue regarding adding new data to the vectors Collection on the fly